src/Adapter/SymfonyCache.php

```diff
@@ -169,6 +178,26 @@ public function cacheAction(string $token, string $type): Response
 
         $path = 'all' === $type ? $this->cacheDir : sprintf('%s/%s', $this->cacheDir, $type);
 
+        /*
+         * NEXT_MAJOR: Remove this if condition and else block.
+         * clean up event dispatcher just before clearing the cache to prevent further events execution
+         */
+        if (null !== $this->eventDispatcher) {
+            foreach ($this->eventDispatcher->getListeners() as $eventName => $eventListenerList) {
+                foreach ($eventListenerList as $listener) {
+                    $listener[0] instanceof EventSubscriberInterface ?
+                        $this->eventDispatcher->removeSubscriber($listener[0]) :
+                        $this->eventDispatcher->removeListener($eventName, [$listener[0], $listener[1]]);
+                }
+            }
+        } else {
+            @trigger_error(
+                'Passing no 9th argument is deprecated since version 3.1 and will be mandatory in 4.0.
+                Pass Symfony\Component\EventDispatcher\EventDispatcherInterface as 9th argument.',
+                E_USER_DEPRECATED
+            );
```

Comment:
Deprecation messages often lack context, let's provide some and get rid of the weird indention in the middle of the message in the process:
```php
<?php
@trigger_error(sprintf(<<<'MESSAGE
'Passing no 9th argument to %s is deprecated since sonata-project/cache-bundle 3.1 and will be mandatory in 4.0.
Pass a Symfony\Component\EventDispatcher\EventDispatcherInterface instance as 9th argument.
MESSAGE,
    __METHOD__
));
```